scripts/bond-calculator.py

```diff
@@ -0,0 +1,148 @@
+"""
+
+log = get_log()
+
+
+def add_one_month(dt: datetime) -> datetime:
+    return dt + relativedelta(months=1)
+
+
+def main() -> None:
+    parser = OptionParser(
+        usage="usage: %prog [options] UTXO or amount",
+        description=DESCRIPTION,
+    )
+    add_base_options(parser)
+    parser.add_option(
+        "-i",
+        "--interest",
+        action="store",
+        type="float",
+        dest="interest",
+        help="Interest rate to use for fidelity bond calculation (instead of interest_rate config)",
+    )
+    parser.add_option(
+        "-e",
+        "--exponent",
+        action="store",
+        type="float",
+        dest="exponent",
+        help="Exponent to use for fidelity bond calculation (instead of bond_value_exponent config)",
+    )
+    parser.add_option(
+        "-m",
+        "--months",
+        action="store",
+        type="int",
+        dest="months",
+        help="For how many months to calculate the fidelity bond values, each month has its own stats (default 12)",
+        default=12,
+    )
+    parser.add_option(
+        "-o",
+        "--orderbook",
+        action="store",
+        type="str",
+        dest="path_to_json",
+        help="Path to the exported orderbook in JSON format",
+    )
+
+    options, args = parser.parse_args()
+    load_program_config(config_path=options.datadir)
+    if len(args) != 1:
+        log.error("Invalid arguments, see --help")
+        sys.exit(EXIT_ARGERROR)
+    if options.interest:
+        jm_single().config.set("POLICY", "interest_rate", str(options.interest))
+    interest = get_interest_rate()
+    if options.exponent:
+        jm_single().config.set("POLICY", "bond_value_exponent", str(options.exponent))
+    if options.path_to_json:
+        try:
+            with open(options.path_to_json, "r", encoding="UTF-8") as orderbook:
+                bond_values = [fb["bond_value"] for fb in loads(orderbook.read())["fidelitybonds"]]
+            percentiles = quantiles(bond_values, n=100, method="inclusive")
+        except FileNotFoundError as exc:
+            log.error(exc)
+            sys.exit(EXIT_ARGERROR)
+        except StatisticsError:
+            log.error("Given JSON file does not contain enough fidelity bonds (at least 2 required)")
+            sys.exit(EXIT_ARGERROR)
+        bonds_sum = sum(bond_values)
+
+    current_time = datetime.now().timestamp()
+    try:
+        utxo_value = amount_to_sat(args[0])
+    except ValueError:
+        # If it's not a valid amount then it has to be a UTXO
+        if jm_single().bc_interface is None:
+            log.error("For calculation based on UTXO access to Bitcoin Core is required")
+            sys.exit(EXIT_FAILURE)
+        success, utxo = utxostr_to_utxo(args[0])
+        if not success:
+            # utxo contains the error message
+            log.error(utxo)
+            sys.exit(EXIT_ARGERROR)
+        utxo_data = jm_single().bc_interface.query_utxo_set(utxo, includeconf=True)[0]
```

Comment:
Just noting that this'll need update after merge of #1295

Reply:
Done in 1637fbb7d7004c4f073bc34ec0e52c9d7a2d510f